### PR TITLE
[eloquent] use remapping tag in urdf/sdf

### PR DESCRIPTION
**Pull request by mikaelarguedas**
Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | None? |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | None, used simulated tb3 to test the implementation of the tag itself https://github.com/ros-simulation/gazebo_ros_pkgs/issues/991#issuecomment-544895833 |

---

## Description of contribution in a few bullet points

- This change is not valid for dashing and older.
- This is the result of a simple search/replace for all argument tags with the new `remapping` tag in SDF.
- Apologies for unrelated whitespace changes

I sanity checked the diff but haven't use this on actual hardware, assuming that the CI here will run all the tests on this change. 
Ran tests locally but had failures, they didn't seem related to this change.

## Alternative or follow-up solution

Pull the turtlebot3 SDF files from upstream once https://github.com/ROBOTIS-GIT/turtlebot3_simulations/pull/101 has been validated / merged

### CODE REVIEW

**Comment by mlherd** [comment]
Thanks for your PR. I am guessing you would like to do these changes becaue of this [PR](https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1011) in gazebo_ros_packages and the warning messages ```[gazebo-1] [WARN] This syntax is deprecated. Use '--ros-args --remap ~/from:=to' instead``` that we have been seeing since Eloquent.

I would say first either confirm that the models that you modified successfully load and work, and the sensor and joint state data are published right topic names (I prefer this method) or wait until Robotis validates and merges your [PR](https://github.com/ROBOTIS-GIT/turtlebot3_simulations/pull/101)

**Comment by mikaelarguedas** [comment]
> Thanks for your PR. I am guessing that you would like to do these changes becaue of this PR in gazebo_ros_packages and the warning messages [gazebo-1] [WARN] This syntax is deprecated. Use '--ros-args --remap ~/from:=to' instead that we have been seeing since Eloquent.

Yes I just wanted to follow-up as the original author of the issue was @mkhansen-intel. I ended up implementing the feature but didnt find the time to propagate it. As I was fixing it in the turtlebot3 packages I figured I'd fix them here as well.
I am not using any of these files in my projects

> I would say let's first either confirm that the models that you modified successfully load and work, and the sensor and joint state data are published right topic names (I prefer this method)

I did do that on the models from the `turtlebot3_gazebo` package. I just posted a summary of the testing at https://github.com/ROBOTIS-GIT/turtlebot3_simulations/pull/101#issuecomment-567877038

For this repo:
TL;DR Got it to work. Tested `ros2 launch nav2_bringup nav2_tb3_simulation_launch.py`
and the resulting topic list is the same.
<details>

```
$ ros2 topic list
/amcl/transition_event
/bt_navigator/transition_event
/camera/camera_info
/camera/image_raw
/clicked_point
/clock
/cmd_vel
/controller_server/transition_event
/global_costmap/costmap
/global_costmap/costmap_updates
/global_costmap/global_costmap/transition_event
/global_costmap/parameter_events
/imu
/initialpose
/joint_states
/local_costmap/costmap
/local_costmap/costmap_updates
/local_costmap/local_costmap/transition_event
/local_costmap/parameter_events
/local_costmap/published_footprint
/local_plan
/map
/map_server/transition_event
/map_updates
/mobile_base/sensors/bumper_pointcloud
/odom
/parameter_events
/particlecloud
/plan
/planner_server/transition_event
/recoveries_server/transition_event
/robot_description
/rosout
/scan
/tf
/tf_static
/waypoint_follower/transition_event
/waypoints
```

</details>

---

#### Issues faced on master
To get it to work for this repo I had to work around multiple issues just to launch master, not sure if they are expected / known:
- navigation2 doesnt build on recent gcc due to warning in dependencies and use of `-Werror`
- dependency on turtlebot3_description is hard still coded in recent models in this repo
- many nodes crash with "UnknownROSArgsError" on master
<details>
```
[robot_state_publisher-3] terminate called after throwing an instance of 'rclcpp::exceptions::UnknownROSArgsError'
[robot_state_publisher-3]   what():  found unknown ROS arguments: '', '', '', '', '', '', '', ''
[amcl-6] terminate called after throwing an instance of 'rclcpp::exceptions::UnknownROSArgsError'
[amcl-6]   what():  found unknown ROS arguments: '', '', '', '', '', '', '', ''
[controller_server-7] terminate called after throwing an instance of 'rclcpp::exceptions::UnknownROSArgsError'
[controller_server-7]   what():  found unknown ROS arguments: '', '', '', '', '', '', '', ''
[planner_server-8] terminate called after throwing an instance of 'rclcpp::exceptions::UnknownROSArgsError'
[planner_server-8]   what():  found unknown ROS arguments: '', '', '', '', '', '', '', ''
[recoveries_server-9] terminate called after throwing an instance of 'rclcpp::exceptions::UnknownROSArgsError'
[recoveries_server-9]   what():  found unknown ROS arguments: '', '', '', '', '', '', '', ''
[bt_navigator-10] terminate called after throwing an instance of 'rclcpp::exceptions::UnknownROSArgsError'
[bt_navigator-10]   what():  found unknown ROS arguments: '', '', '', '', '', '', '', ''
[waypoint_follower-11] terminate called after throwing an instance of 'rclcpp::exceptions::UnknownROSArgsError'
[waypoint_follower-11]   what():  found unknown ROS arguments: '', '', '', '', '', '', '', ''
[map_server-5] [ERROR] [1576832626.436399616] [map_server]: found unknown ROS arguments: '', '', '', '', '', '', '', ''
```
</details>
  - reverting https://github.com/ros2/rclcpp/pull/951 seems to fix it

- doesn't find the model in nav2_bringup:
```
Error Code: 11 Msg: Unable to find uri[model://turtlebot3_world]
```
-
  - preprending `/tmp/tb3_demo_ws/install/nav2_bringup/share/nav2_bringup/worlds` to `GAZEBO_MODEL_PATHS` did *not* fix it
  - preprending `/tmp/tb3_demo_ws/install/turtlebot3_gazebo/share/turtlebot3_gazebo/models` to `GAZEBO_MODEL_PATHS` seems to fix it.

Same topic list before and after the PR:

Then I ran out of time and didnt test the changes in the `nav2_system_tests` files

**Comment by SteveMacenski** [comment]
This looks fine from a glance and I don't see a reason to suspect another problem, but I would like to see this working on someone's side before merging. The issues you mention I haven't run into personally, but may have also been an old commit on master branch.  

I know its not great practice to develop off of a released distribution branch, but most of your problems are from master and you should be able to apply these commits to `eloquent` binaries and show they work.

**Comment by mikaelarguedas** [comment]
This seem to work on `eloquent-devel` too:

Place navigation2 + turtlebot3_description in a workspace overlaying eloquent binaries
```
export NAV2_WS=`pwd`
cd $NAV2_WS/src/navigation2 && git cherry-pick 3b280daf676e0aaab5b7300af612608d2a4b647b && cd $NAV2_WS
colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=ON
export GAZEBO_MODEL_PATH=$NAV2_WS/src/navigation2/nav2_system_tests/models/:$GAZEBO_MODEL_PATH
ros2 launch nav2_bringup nav2_tb3_simulation_launch.py
```

```
# ros2 topic list
/amcl/transition_event
/bt_navigator/transition_event
/camera/camera_info
/camera/image_raw
/clicked_point
/clock
/cmd_vel
/controller_server/transition_event
/global_costmap/costmap
/global_costmap/costmap_updates
/global_costmap/global_costmap/transition_event
/global_costmap/parameter_events
/global_costmap/rosout
/imu
/initialpose
/joint_states
/local_costmap/costmap
/local_costmap/costmap_updates
/local_costmap/local_costmap/transition_event
/local_costmap/parameter_events
/local_costmap/published_footprint
/local_costmap/rosout
/local_plan
/map
/map_server/transition_event
/map_updates
/mobile_base/sensors/bumper_pointcloud
/odom
/parameter_events
/particlecloud
/plan
/planner_server/transition_event
/recoveries_server/transition_event
/robot_description
/rosout
/scan
/tf
/tf_static
/waypoint_follower/transition_event
/waypoints
```

**Comment by SteveMacenski** [comment]
Sounds good enough to me.
